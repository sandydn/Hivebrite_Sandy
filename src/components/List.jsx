import React from 'react';
import styled from 'styled-components';
import Card from './Card';

const ListWrapper = styled.div`
    height: 510px;
    overflow-y: scroll;
    overflow-x: hidden;
`;

const ButtonMore = styled.button`
    position: absolute;
    bottom: 20px;
    font-size: 1em;
    margin: 0em 2.95em;
    padding: 0.15em 1em;
    border: 2px solid #8baee6;
    border-radius: 3px;
    color: #84a3d6;
`;

const CardNotFound = styled.div`
    width: 215px;
    padding: 20px;
    margin: 25px 45px 0px 3px;
`;

const TextNotFound = styled.h2`
    margin: 10px 25px;
    font-size: 1.1rem;
    width: 140px;
`;

const List = props => {
    const {cities, selectedCity, handleClick, handleMore} = props;

    return (
        <ListWrapper >
                {cities.length > 0 ? 
                    cities.map(city => <Card key={city.rank} cityData={city} selectedCity={selectedCity} handleClick={handleClick}/>)
                    :
                    <CardNotFound>
                        <TextNotFound>Oops not found ...</TextNotFound>
                    </CardNotFound>
                }
           <ButtonMore data-jest='button-more' onClick={handleMore}>Shom Me More</ButtonMore>
        </ListWrapper>
    );
};

export default List;
