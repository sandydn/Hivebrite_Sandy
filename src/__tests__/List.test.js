import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import List from '../components/List.jsx';

describe('List', () => {
  const onHandleClickMock = jest.fn();
  const props = {
      cities : [
        {
            "city": "New York", 
            "growth_from_2000_to_2013": "4.8%", 
            "latitude": 40.7127837, 
            "longitude": -74.0059413, 
            "population": "8405837", 
            "rank": "1", 
            "state": "New York"
        }, 
        {
            "city": "Los Angeles", 
            "growth_from_2000_to_2013": "4.8%", 
            "latitude": 34.0522342, 
            "longitude": -118.2436849, 
            "population": "3884307", 
            "rank": "2", 
            "state": "California"
        }, 
        {
            "city": "Chicago", 
            "growth_from_2000_to_2013": "-6.1%", 
            "latitude": 41.8781136, 
            "longitude": -87.6297982, 
            "population": "2718782", 
            "rank": "3", 
            "state": "Illinois"
        }
    ],
    selectedCity: {
        city: "New York",
        growth_from_2000_to_2013: "4.8%",
        latitude: 40.7127837,
        longitude: -74.0059413,
        population: "8405837",
        rank: "1",
        state: "New York"
    },
    handleMore: () => onHandleClickMock('Arg')
  };
    
    const wrapper = shallow(<List {...props} />);

    describe('Should pass on start.', () => {
        it('Should render correctly.', () => {
            const tree = renderer
                .create(<List {...props} />)
                .toJSON();
        expect(tree).toMatchSnapshot();
        });

        it('Should call "handleMore" when clicking on the button.', () => {
            wrapper.find("[data-jest='button-more']").simulate('click');
        expect(onHandleClickMock).toHaveBeenCalled();
        });
    });
});