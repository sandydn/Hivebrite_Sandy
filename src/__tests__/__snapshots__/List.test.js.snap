// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Should pass on start. Should render correctly. 1`] = `
<div
  className="sc-Axmtr gTuKYw"
>
  <div
    className="sc-AxjAm jgLRTK"
    data-jest="card"
    onClick={[Function]}
  >
    <div
      className="sc-AxirZ eTxVFc"
    >
      <h2
        className="sc-AxhCb jAlGnI"
      >
        New York
      </h2>
      <h3
        className="sc-AxhUy jSGLIL"
      >
        New York
      </h3>
    </div>
    <div
      className="sc-AxirZ eTxVFc"
    >
      <div
        className="sc-AxiKw kKjxUC"
      >
        <img
          alt="population"
          className="sc-AxgMl jLPpZy"
          src="https://img.icons8.com/fluent/25/000000/conference-call.png"
        />
        <h3
          className="sc-AxheI kLPUmH"
        >
          8405837
        </h3>
      </div>
      <div
        className="sc-AxiKw kKjxUC"
      >
        <img
          className="sc-AxgMl jLPpZy"
          src="https://img.icons8.com/fluent/35/000000/up.png"
        />
        <h3
          className="sc-AxheI kLPUmH"
        >
          4.8%
        </h3>
      </div>
    </div>
  </div>
  <div
    className="sc-AxjAm dgXrKn"
    data-jest="card"
    onClick={[Function]}
  >
    <div
      className="sc-AxirZ eTxVFc"
    >
      <h2
        className="sc-AxhCb jAlGnI"
      >
        Los Angeles
      </h2>
      <h3
        className="sc-AxhUy jSGLIL"
      >
        California
      </h3>
    </div>
    <div
      className="sc-AxirZ eTxVFc"
    >
      <div
        className="sc-AxiKw kKjxUC"
      >
        <img
          alt="population"
          className="sc-AxgMl jLPpZy"
          src="https://img.icons8.com/fluent/25/000000/conference-call.png"
        />
        <h3
          className="sc-AxheI kLPUmH"
        >
          3884307
        </h3>
      </div>
      <div
        className="sc-AxiKw kKjxUC"
      >
        <img
          className="sc-AxgMl jLPpZy"
          src="https://img.icons8.com/fluent/35/000000/up.png"
        />
        <h3
          className="sc-AxheI kLPUmH"
        >
          4.8%
        </h3>
      </div>
    </div>
  </div>
  <div
    className="sc-AxjAm dgXrKn"
    data-jest="card"
    onClick={[Function]}
  >
    <div
      className="sc-AxirZ eTxVFc"
    >
      <h2
        className="sc-AxhCb jAlGnI"
      >
        Chicago
      </h2>
      <h3
        className="sc-AxhUy jSGLIL"
      >
        Illinois
      </h3>
    </div>
    <div
      className="sc-AxirZ eTxVFc"
    >
      <div
        className="sc-AxiKw kKjxUC"
      >
        <img
          alt="population"
          className="sc-AxgMl jLPpZy"
          src="https://img.icons8.com/fluent/25/000000/conference-call.png"
        />
        <h3
          className="sc-AxheI kLPUmH"
        >
          2718782
        </h3>
      </div>
      <div
        className="sc-AxiKw kKjxUC"
      >
        <img
          className="sc-AxgMl jLPpZy"
          src="https://img.icons8.com/fluent/35/000000/down.png"
        />
        <h3
          className="sc-AxheI kLPUmH"
        >
          -6.1%
        </h3>
      </div>
    </div>
  </div>
  <button
    className="sc-AxmLO kMwRHj"
    data-jest="button-more"
    onClick={[Function]}
  >
    Shom Me More
  </button>
</div>
`;
